### DISCUSSION TITLE
SQLite with ANSI dialect, indexes?

### DISCUSSION THREAD

**Discussion by martinpaljak** [micronaut-projects/micronaut-data#1409] [Q&A]
Is SQLite supposed to be supported? I'm trying to make it work and it is _mostly_ OK for very simple schemas with some hand-tuning. Even an index is added, but just one from two indexes specified via jakarta @Indexes annotation.

```
@Indexes(value = {
        @Index(name = "index_idcode", columns = "idcode"),
        @Index(name = "index_hash", columns = "sha256")
})
```
Produces 

```
CREATE TABLE IF NOT EXISTS "cached_certificate" ("id" UUID,"idcode" BIGINT NOT NULL,"sha256" BLOB NOT NULL,"certificate" BLOB NOT NULL);
CREATE INDEX index_idcode ON "cached_certificate" (idcode);
```

What should be done to have an actual "sqlite" dialect in micronaut-data?

**Comment by dstepanov** [comment]
What is the difference? We recommend using something like flyway to set up your DB schema.

**Comment by martinpaljak** [comment]
I don't want to use flyway, the database is supposed to be created during startup and there's no migration necessary. I just want to keep all the information in Java code as annotations.

Is there a "standard" way to open a bounty for "full sqlite support" in micronaut-data?

**Reply by dstepanov** [reply]
You can create an issue, not sure what you mean by bounty.
Maybe the solution would be to modify the actual query for the ANSI dialect.

**Reply by martinpaljak** [reply]
I'd offer a small bounty for possible developer, if there was a  "standard" way of doing it?
